src/Octoshift/GithubClient.cs

```diff
@@ -186,8 +204,24 @@ private string GetNextUrl(KeyValuePair<string, IEnumerable<string>>[] headers)
         private string ExtractLinkHeader(IEnumerable<KeyValuePair<string, IEnumerable<string>>> headers) =>
             ExtractHeaderValue("Link", headers);
 
+        private int RateLimitRemaining(IEnumerable<KeyValuePair<string, IEnumerable<string>>> headers) =>
+            int.Parse(ExtractHeaderValue("X-RateLimit-Remaining", headers) ?? "5000");
+
+        private long RateLimitReset(IEnumerable<KeyValuePair<string, IEnumerable<string>>> headers) =>
+            long.Parse(ExtractHeaderValue("X-RateLimit-Reset", headers) ?? DateTimeOffset.Now.ToUnixTimeSeconds().ToString());
+
         private string ExtractHeaderValue(string key, IEnumerable<KeyValuePair<string, IEnumerable<string>>> headers) =>
             headers.SingleOrDefault(kvp => kvp.Key.Equals(key, StringComparison.OrdinalIgnoreCase)).Value?.FirstOrDefault();
 
+        private void CheckForRetryDelay(IEnumerable<KeyValuePair<string, IEnumerable<string>>> headers)
+        {
+            if (RateLimitRemaining(headers) == 0)
+            {
+                var resetUnixSeconds = RateLimitReset(headers);
+                var currentUnixSeconds = _dateTimeProvider.CurrentUnixTimeSeconds();
+
+                _retryDelay = (resetUnixSeconds - currentUnixSeconds) * 1000.0;
```

Comment:
here also I would create a `const` for 1000, and make it clear that it's milliseconds (or whatever).